analyzers/tests/SonarAnalyzer.UnitTest/Trackers/ConstantValueFinderTest.cs

```diff
@@ -57,5 +57,34 @@ public int Method()
             var finder = new CSharpConstantValueFinder(compilation.GetSemanticModel(tree));
             finder.FindConstant(returnExpression).Should().Be(42);
         }
+
+        [TestMethod]
+        public void WrongCompilationBeingUsed()
+        {
+            const string code1 = @"
+public partial class Sample
```

Comment:
using `partial class` and the same name makes the test hard to follow, because the two partial classes are in two different solutions, so they're totally different.

IMO for the sake of testing this scenario you can have just two classes `Foo` and `Bar`  and pass the wrong Compilation to the class, and it will be easy to understand what this test is doing